flytekit/core/map_task.py

```diff
@@ -1,72 +1,168 @@
+    ):
         collection_interface = transform_interface_to_list_interface(tk.python_interface)
-        name = "mapper_" + tk.name
+        name = f"{tk._task_function.__module__}.mapper_{tk._task_function.__name__}"
         self._run_task = tk
+        self._max_concurrency = concurrency
+        self._min_success_ratio = min_success_ratio
+        self._array_task_interface = tk.python_interface
         super().__init__(
             name=name,
             interface=collection_interface,
             metadata=metadata,
-            task_type="map_task",
+            task_type="container_array",
             task_config=None,
+            task_type_version=1,
             **kwargs,
         )
 
+    def get_command(self, settings: SerializationSettings) -> List[str]:
+        return [
+            "pyflyte-map-execute",
+            "--task-module",
+            self._run_task._task_function.__module__,
+            "--task-name",
+            f"{self._run_task._task_function.__name__}",
+            "--inputs",
+            "{{.input}}",
+            "--output-prefix",
+            "{{.outputPrefix}}",
+            "--raw-output-data-prefix",
+            "{{.rawOutputDataPrefix}}",
+        ]
+
+    def get_container(self, settings: SerializationSettings) -> Container:
+        env = {**settings.env, **self.environment} if self.environment else settings.env
+        return _get_container_definition(
+            image=get_registerable_container_image(None, settings.image_config),
+            command=[],
+            args=self.get_command(settings=settings),
+            data_loading_config=None,
+            environment=env,
+        )
+
     @property
     def run_task(self) -> PythonTask:
         return self._run_task
 
     def execute(self, **kwargs) -> Any:
-        all_types_are_collection = True
-        any_key = None
-        for k, v in self._run_task.interface.inputs.items():
-            any_key = k
-            if v.type.collection_type is None:
-                all_types_are_collection = False
-                break
+        ctx = FlyteContext.current_context()
+        if ctx.execution_state and ctx.execution_state.mode == ExecutionState.Mode.TASK_EXECUTION:
+            return self._execute_map_task(ctx, **kwargs)
 
-        # If all types are collection we can just handle the call as a pass through
-        if all_types_are_collection:
-            return self._run_task.execute(**kwargs)
+        return self._raw_execute(**kwargs)
+
+    @staticmethod
+    def _compute_array_job_index() -> int:
+        """
+        Computes the absolute index of the current array job. This is determined by summing the compute-environment-specific
+        environment variable and the offset (if one's set). The offset will be set and used when the user request that the
+        job runs in a number of slots less than the size of the input.
+        """
+        offset = 0
+        if os.environ.get("BATCH_JOB_ARRAY_INDEX_OFFSET"):
```

Comment:
is this only for AWS Batch>?

Reply:
if so can we add that to the docs?

Reply:
nope! k8s too. see the double env var lookup below https://github.com/flyteorg/flytekit/pull/405/files#diff-38030c87fd7703a4c95a033167b2ce8efa1d3b61824e891eb063bb1337def271R88 this gets the appropriate index env var name (k8s or batch specific) and then looks that up